test/mocks/network/mocks.cc

```diff
@@ -97,8 +97,10 @@ MockUdpListenerCallbacks::~MockUdpListenerCallbacks() = default;
 MockDrainDecision::MockDrainDecision() = default;
 MockDrainDecision::~MockDrainDecision() = default;
 
-MockListenerFilter::MockListenerFilter() = default;
-MockListenerFilter::~MockListenerFilter() = default;
+MockListenerFilter::MockListenerFilter() {
+  // ON_CALL(*this, die_()).WillByDefault(testing::Invoke([](){}));
```

Comment:
I don't really know why ON_CALL + NiceMock reports uninterested call.
I add explicit EXPECT_CALL in each usage. That's a tech debt.
I will investigate later

Reply:
It doesn't work in the destructor of a mock. We do this all over the place. Please delete this comment.

Reply:
Sad...`During the constructor or destructor of MockFoo, the mock object is not nice or strict`